File: src/solutions/73.cpp
#include "../lib/euler.hpp"
#include <iostream>
using namespace std;

int main() {
    int N = 12000;
    int ans = 0;
    for (int num = 2; num < N/3 + 1; ++num) {
        int den = 2*num + 1;
        while (double(num)/den > double(1)/3) {
            if (gcd(num, den) == 1) {
                //cout << num << " " << den << endl;
                ++ans;
            }
            ++den;
        }
    }
    cout << ans << endl;
}

File: archive/src/solutions/73.py
from fractions import Fraction as fr, gcd
ans = 0
n = 12001

#1/3 < p/q < 1/2
#q < 12001
#p > 1/3*12001  (min value of p)
#p < 1/2*12001 (max value of p) 

for num in xrange(1, n/3+2):
    if num % 100 == 0:
        print 'num =', num
    den = 2*num + 1
    while float(num)/den > fr(1, 3):
        if gcd(num, den) == 1:
            ans += 1
        den += 1
           
print ans
